sql/core/src/test/scala/org/apache/spark/sql/DateFunctionsSuite.scala

```diff
@@ -475,6 +1164,45 @@ class DateFunctionsSuite extends QueryTest with SharedSQLContext {
       Row(ts1.getTime / 1000L), Row(ts2.getTime / 1000L)))
   }
 
+  test("to_unix_timestamp with session local timezone") {
```

Comment:
The problem is, except this suite, all the changes you made to the tests are just fixed existing tests to fit the timezone stuff. You add all the new tests in this suite as end-to-end tests, which is not good. We should add new tests in `DateTimeUtilsSuite` as unit tests.